Zend/zend_operators.c

```diff
@@ -742,6 +754,12 @@ ZEND_API zend_long ZEND_FASTCALL _zval_get_long_func(zval *op) /* {{{ */
 		case IS_DOUBLE:
 			return zend_dval_to_lval(Z_DVAL_P(op));
 		case IS_STRING:
+			if (!silent) {
+				if (is_numeric_string(Z_STRVAL_P(op), Z_STRLEN_P(op), NULL, NULL, -1)==0) {
+					zend_error(E_WARNING, "A non-numeric value encountered");
+					return 0;
+				}
+			}
```

Comment:
Supporting floats and their notation (scientific) would make PHP more consistent. We currently support this for long parameters in argument parsing, e.g.

    $ php -r 'var_dump(intdiv("1e3", 1));'
    int(1000)

Yet due to the use of `strtol()` here, we don't for the integer operators, nor for `(int)` and `intval()`. So this means that `3e3` is considered to be `3` for some integer operations and `3000` for others, which is not ideal.

Would it cause backwards-compatibility issues if we started allowing floats here? I don't think it would.